libtsuba/src/RDGPartHeader.cpp

```diff
@@ -26,6 +26,18 @@ const char* kPartProperyMetaKey = "kg.v1.part_property_meta";
 
 // special partition property names
 
+katana::Result<void>
+CopyProperty(
+    tsuba::PropStorageInfo* prop, const katana::Uri& old_location,
+    const katana::Uri& new_location) {
+  katana::Uri old_path = old_location.Join(prop->path());
+  katana::Uri new_path = new_location.Join(prop->path());
+  tsuba::FileView fv;
+
+  KATANA_CHECKED(fv.Bind(old_path.string(), true));
+  return tsuba::FileStore(new_path.string(), fv.ptr<uint8_t>(), fv.size());
```

Comment:
This is good and property-at-a-time is probably sufficient for the long term.  It seems like there is a GCS REST API to copy a storage object without reading it.
https://cloud.google.com/storage/docs/copying-renaming-moving-objects#copy-object-json

Reply:
Totally; good to keep this in mind. I was worried that this wouldn't work with the way we have it implemented via the aws sdk, so opted for something that was guaranteed to work.